### ISSUE TITLE
deCONZ: failed logins registered when only surfing Phoscon part in ingress (no need to provide password)

### ISSUE THREAD

**Issue by bcutter** [home-assistant/addons#3595]
### Describe the issue you are experiencing

Only surfing `/core_deconz/ingress` and selecting "Phoscon" ("deCONZ" works just fine) creates failed logins for the client. When using HTTP security (ban module on frequent failed logins) this immediately leads to a blocked client in `ip_bans.yaml` which hurts quite a bit (need to clean the file and restart HA Core):
```
xxx.xxx.xxx.xxx:
  banned_at: '2024-05-10T15:38:43.146448+00:00'
```

I can reproduce this issue with different clients (Win 11 in Firefox, iOS with HA Companion app, doesn't really matter).

It is sufficient to only load the Phoscon part, no need to enter any password at all. Some automatic API request seem to fail.

Assumption: As probably most users don't use HTTP security this issue has not been discovered so far.

### What type of installation are you running?

Home Assistant OS

### Which operating system are you running on?

Home Assistant Operating System

### Which add-on are you reporting an issue with?

deCONZ

### What is the version of the add-on?

6.23.0

### Steps to reproduce the issue

1. Load deCONZ in ingress (`/core_deconz/ingress`)
2. Select "Phoscon"
3. Wait and see the failed logins

## System Information

version | core-2023.4.6
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.10.10
os_name | Linux
os_version | 6.1.73-haos-raspi
arch | aarch64
timezone | Europe/Berlin
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
GitHub API Calls Remaining | 5000
Installed Version | 1.32.1
Stage | running
Available Repositories | 1478
Downloaded Repositories | 85
HACS Data | ok

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | false
-- | --
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 12.2
-- | --
update_channel | stable
supervisor_version | supervisor-2024.05.1
agent_version | 1.6.0
docker_version | 25.0.5
healthy | true
supported | true
board | rpi4-64
supervisor_api | ok
version_api | ok

</details>

### Anything in the Supervisor logs that might be useful for us?

```txt
No, nothing specially related to deCONZ. But see HA logs below.

I see one entry which might be related to "a0d7b954_vscode" used at the same time and having some more log entries before (please check with port and URL if this is deCONZ or VS Code related):

2024-05-10 17:38:26.680 ERROR (MainThread) [supervisor.api.ingress] Stream error with http://172.30.33.2:8099/pwa/language/de/index-de.json: Cannot write to closing transport
```

### Anything in the add-on logs that might be useful for us?

```txt
Unfortunately I could not grab addon logs, as the productive log data was too much so the ones from around the last incident was not visible anymore.

Anyway, this is what is logged in HA log:

2024-05-10 17:38:27.691 WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from client-name.local (xxx.xxx.xxx.xxx). Requested URL: '/api/config?_=1715355507074'. (Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:125.0) Gecko/20100101 Firefox/125.0)
2024-05-10 17:38:32.428 WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from client-name.local (xxx.xxx.xxx.xxx). Requested URL: '/api/config?_=1715355507075'. (Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:125.0) Gecko/20100101 Firefox/125.0)
2024-05-10 17:38:38.146 WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from client-name.local (xxx.xxx.xxx.xxx). Requested URL: '/api/config?_=1715355507076'. (Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:125.0) Gecko/20100101 Firefox/125.0)
2024-05-10 17:38:42.462 WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from client-name.local (xxx.xxx.xxx.xxx). Requested URL: '/api/config?_=1715355507077'. (Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:125.0) Gecko/20100101 Firefox/125.0)
2024-05-10 17:38:43.126 WARNING (MainThread) [homeassistant.components.http.ban] Login attempt or request with invalid authentication from client-name.local (xxx.xxx.xxx.xxx). Requested URL: '/api/config?_=1715355507078'. (Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:125.0) Gecko/20100101 Firefox/125.0)
2024-05-10 17:38:43.129 WARNING (MainThread) [homeassistant.components.http.ban] Banned IP xxx.xxx.xxx.xxx for too many login attempts
```

### Additional information

Possibly related to the Phoscon bug introduced with addon version 6.21.0 (https://github.com/home-assistant/addons/pull/3226#issuecomment-1737293462), being reverted in 6.22.0 and "somehow" fixed (worked around?) in 6.23.0.

During that adventure the Phoscon page was in the spotlight of issues already. Maybe this issue is a leftover or a side effect of the fix.

**Comment by mirml** [comment]
I've got same problem ...

**Comment by raolss** [comment]
Similar issue, I get this problem when starting Phoscon from Campanion app but not from Chrome on windows

**Comment by bcutter** [comment]
**Test 1:**
Tried with Google Chrome on Android - same issue.

It really doesn't matter which browser/app is used.

Really starts to annoy me as hell as it makes it **impossible** to use Phoscon ~at all~ in Home Assistant!

**Test 2:**
Tried to directly access Phoscon by using http://servername.domain:40850 in the browser. This is working. So it seems to be an Ingress related issue.

**Comment by mirml** [comment]
Yes it works only with local IP, but not with my domain name.

**Comment by bcutter** [comment]
Still broken. And no one cares. Great perspective for deCONZ users in Home Assistant.

**Comment by mirml** [comment]
Yes still broken. It's really strange that there aren't more complaints. I know for myself that the next coordinator will not be a conbee.

**Comment by DanielHabenicht** [comment]
Same problem here, successfully locked me out with all my devices somehow :)

**Comment by bcutter** [comment]
The lack of interest from HA side (because of ZHA) and the lack of man power (and understanding how important the HA addon is in their chain and for their user base) from Dresden will very likely be the reason for this issue to never get solved. Same drama like with releases (we're now half a year behind the latest release...).

For this case both is needed: knowledge of Phoscon and of Ingress (HA side).

Feel free to prove me wrong.

**Comment by mirml** [comment]
Very sad.

**Comment by mirml** [comment]
https://github.com/home-assistant/addons/pull/3708

**Comment by bcutter** [comment]
> Upgrade deConz to v2.27.6  :)
> 
> https://github.com/home-assistant/addons/pull/3708

Interesting. Where's the proof it fixes the issue? Any specific change or what are you referring to?

And recent PRs for the 2.27.6 release have been rejected so far, so I don't see a quick solution yet.

**Comment by bcutter** [comment]
Still exists for me with deCONZ v2.28.1 :-(

![grafik](https://github.com/user-attachments/assets/4c5727c5-2b41-4c92-88f7-fdebe34a4fd6)

As this issue sits right between the two competence fields of both involved parties (HA team on the Ingress side and Dresden Elektronik team on the deCONZ side), I doubt this will be fixed or even looked into until they found an agreement on the deCONZ addon support topic.

**Comment by mirml** [comment]
Yes, I ordered sonoff zigbee dongle E and I am moving my zigbee network to it right now.

**Comment by bcutter** [comment]
Massive issue, definitely needing a fix. Phoscon not usable in HA ingress at all with http settings enabled.

**Comment by bcutter** [comment]
Still screwed up. And so many people care about deCONZ in HA, wow.

**Comment by mraude01** [comment]
Same probleme here I will send the Raspbee back an switch to another Zigbee solution for HA

**Comment by bcutter** [comment]
Nope. Alive. And annoying on a daily basis. deCONZ is useless (unusable) in Ingress.

**Comment by RolandHome** [comment]
Problem besteht weiter, ich habe meinen neuen ConBee_III zurückgegeben. Warum soll ich einer Firma weiter Geld geben wenn es keinen Support gibt. Ich nutze weiter meinen ConBee_II mit dem gleichen Problem (nur um mal zu schauen wie lange das noch dauert um ein so "kleines" Problem zu lösen). Ich glaube in Dresden wird nur als Hobby in der Garage gebastelt. Neue und gute Lösung für mich: SLZB-06. Sehr schade, ich hätte lieber von einer deutschen Firma gekauft...............

**Comment by bcutter** [comment]
Dear bot you are absolutely annoying and useless (at least a nice reminder we still suffer from an issue) because no one is taking care to fix this issue.

Because of this it will stay open until someone will fix it. So we will play this stale -> unstable game like forever.

**Comment by agners** [comment]
From what I can tell, this is a problem of the Phoscon web app. It seems to poll the absolute path `/api/config` on the origin host/port, which is where Home Assistant Core http server is at. 

![Image](https://github.com/user-attachments/assets/b5eda0bc-5703-41a3-ad66-edb39f826b61)

Phoscon should not poll that end point, but instead the relative endpoint.... I don't think much can be done from the add-on side.

**Comment by bcutter** [comment]
Interesting discovery. So when was this behaviour introduced and how couldn't be reverted or modified in the way you mention it?

Some deCONZ expert is needed very likely.

**Comment by vgisbert** [comment]
Same problem

**Comment by bcutter** [comment]
An old thing:

https://github.com/home-assistant/addons/issues/2658

**Comment by mattreim** [comment]
Hello @agners, from my perspective, the API key is missing in this path: [`/api/<apikey>/config`](https://dresden-elektronik.github.io/deconz-rest-doc/endpoints/configuration/#get-configuration). It is not stored by Home Assistant or read from the integration, so the connection will always be unauthorized. The deCONZ integration receives a fixed API key the first time it connects to the deCONZ REST API. Maybe it will help.

**Comment by mattreim** [comment]
Hello everyone, the bug has now been fixed with version [2.31.1-beta](https://github.com/dresden-elektronik/deconz-rest-plugin/releases/tag/v2.31.1-beta) and will soon be available to everyone. So just a little patience. 

A big thank you to @manup.

**Comment by bcutter** [comment]
Stable v2.31.2 containing the fix is available now.

Who will file a PR to update the deCONZ addon?

https://github.com/dresden-elektronik/deconz-rest-plugin/releases/tag/v2.31.2

**Comment by mattreim** [comment]
The [PR ](https://github.com/home-assistant/addons/pull/4161) is already open but we still have to wait.

**Comment by bcutter** [comment]
Great, thanks 👍

**Comment by mattreim** [comment]
@bcutter, has the error been resolved for you now?

**Comment by bcutter** [comment]
Didn't have the chance to update the addon yet. Will report back once updated and tested.

**Comment by bcutter** [comment]
Login still fails using Ingress:

<img width="755" height="349" alt="Image" src="https://github.com/user-attachments/assets/edbe27d2-959f-479f-869b-132b5190be66" />

It's a cashing issue as still the old UI was presented:

<img width="1147" height="259" alt="Image" src="https://github.com/user-attachments/assets/5549283e-b14a-4c07-924b-7972729f46c0" />

Using a private browsing tab the new frontend was shown:

<img width="870" height="380" alt="Image" src="https://github.com/user-attachments/assets/ebceee2d-b020-47ed-a058-ac8cf38e4d62" />

~~Solution: reload the Phoscon/deCONZ select page frame (Firefox). Read it somewhere few days ago, couldn't find it.~~

Unfortunately, that only worked for the overview page. Logging in using Phoscon still fails, something must be in browser cash here still.

<img width="825" height="395" alt="Image" src="https://github.com/user-attachments/assets/cbbf1ab9-68de-49f7-a51a-7a086b0584b8" />

deCONZ addon log when hitting the login button:

```
172.30.32.2 - delight [28/Oct/2025:01:00:41 +0100] "POST /api/ HTTP/1.1" 431 5 "http://my.home-assistant.local:8123/api/hassio_ingress/XXXXXXXXXXXXXXXXXXXXXXXXXX/pwa/login2.html?1761609619085" "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:144.0) Gecko/20100101 Firefox/144.0"
```

**Comment by mattreim** [comment]
Does logging in outside of HA work? You should also clear the browser cache beforehand.

**Comment by bcutter** [comment]
Yes, flawlessly. Also from inside HA using a private browsing tab in the same browser.
I guess it's a browser cache issue. Just no idea how to precisely solve that without purging all browser cache which is... not a good idea.

**Comment by mattreim** [comment]
https://github.com/home-assistant/addons/pull/4161#issuecomment-3438115999
> Hm, I realized that Ctrl+F5 does not force a refresh of iframes, so the browser simply showed the cached version. Right clicking inside the ingress iframe and refreshing the frame specifically made the new page load (this is available in Firefox as well as in Chrome) 👍 .
> 
> It seems to me that browser still should request the content more aggressively, but maybe the ingress proxy causes some issues here. But anyhow, all this is unrelated to this PR.

As already mentioned here it is an HA problem and not a deCONZ/Phoscon problem. Unfortunately, the only thing that helps is clearing the browser cache every now and then.

**Comment by bcutter** [comment]
Selectively purging the cache for the whole website HA is running on (domain/subdomain) made the issue disappear. Login possible, surfing does not trigger a failed login attempt anymore (at least I think so as I meanwhile turned off my auto lock mechanisms because this issue existed for a while - 1.5 years). Glad this is finally sorted out. Thanks a lot @mattreim 🥳

**Comment by agners** [comment]
Just FYI, there is a reload frame option which should bypass the disk cache (similar option exists for Chrome)

<img width="789" height="593" alt="Image" src="https://github.com/user-attachments/assets/afa8adf0-eff6-430d-adde-b3205c1de581" />

But anyhow, glad the latest deCONZ changes fixed the login issue 👍

**Comment by bcutter** [comment]
Yes, but that frame reload way did not do the trick in Firefox.
